ipn/ipnlocal/local.go

```diff
@@ -2769,13 +2779,28 @@ func (b *LocalBackend) DebugPickNewDERP() error {
 //
 // b.mu must not be held.
 func (b *LocalBackend) send(n ipn.Notify) {
+	b.sendTo(n, nil)
+}
+
+// notificationTarget represents an identifiable notification recipient.
+// In most cases, it is an [ipnauth.Actor] (and may also implement an [ipnauth.Session]).
+type notificationTarget interface {
+	// UserID returns an OS-specific UID of the receiver.
+	UserID() ipn.WindowsUserID
+}
```

Comment:
why is this not a SessionID (or PIDHandle) that's being notified? Why is this generic type using a WindowsUserID?

Reply:
I changed it to
```go
// notificationTarget describes a notification recipient.
// A zero value or a nil pointer are valid and indicate that
// the notification should be broadcast to all active [watchSession]s.
type notificationTarget struct {
	// UserID is the OS-specific UID of the target user.
	// If empty, the notification is not user-specific and
	// will be broadcast to all connected users.
	// TODO(nickkhyl): make this field cross-platform rather
	// than Windows-specific.
	UserID ipn.WindowsUserID
	// ClientID identifies a client that should be the exclusive recipient
	// of the notification. A zero value indicates that notification should
	// be sent to all sessions of the specified user.
	ClientID ipnauth.ClientID
}
```

Re `WindowsUserID`: Currently, `LocalBackend` does not clearly distinguish between users on platforms other than Windows. For instance, on Windows, Tailscale profiles are owned by specific users, but there is no concept of profile ownership[*] on other operating systems. Therefore, the existing `ipnauth.ConnIdentity` type has methods like `(*ConnIdentity).WindowsUserID() ipn.WindowsUserID`, and `LocalBackend` had a `currentUser ipnauth.WindowsToken` field until recently, and so on. I don't think `WindowsUserID` should exist in the long term, but I'd like to make its removal more intentional and move certain code from `ipn/ipnlocal` and `ipn/localapi` to `ipn/ipnauth` first.

[*] We do have the concept of an "OperatorUser" on Unix-like platforms, but it's implemented differently, has a slightly different meaning in the current permission model, and uses a username instead of a UID.